Zoo/MainZoo.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Zoo
+{
+    public delegate void notify(int time);
+    public class MainZoo
+    {
+        const int workDay = 120;
+        const int tourTime = 10;
+
+        static ZooManager Manager = ZooManager.Instance;
+        public static int time = 0;
+
+        private List<Animal> animals;
+        private List<Worker> workers;
+        private Dictionary<int, List<Worker>> workersDictStart;
+        private Dictionary<int, List<Worker>> workersDictFinish;
+        private List<Tour> tours;
+        private int tourId;
+        private int zooIncome;
+
+
+        public event notify zooEvents;
+
+        public MainZoo(List<Animal> animals, List<Worker> workers)
+        {
+            if (CheckZoo(animals))
+            {
+                this.animals = animals;
+                tourId = 1;
+                this.workers = workers;
+                workersDictStart = new Dictionary<int, List<Worker>>();
+                workersDictFinish = new Dictionary<int, List<Worker>>();
+                tours = new List<Tour>();
+                AddWorkersToDicts();
+                zooEvents += new notify(AddAndRemoveTours);
+                zooEvents += new notify(PrintStartAndFinish);
+                zooIncome = 0;
+                Manager.LogEvents("Zoo created.");
+            }
+            else
+                Manager.LogEvents("Can't create Zoo.");
+        }
+
+        public bool CheckZoo(List<Animal> animalArr)
```

Comment:
Right now you only check animal type count, but in the future someone might want to add another validation, making this function hella long or requiring changes. What do we do to prevent it? we split functions by their functionality- function to validate animal type count and a function to validate the whole zoo.